src/pages/uxp-api/reference-js/Modules/uxp/XMP/index.md

```diff
@@ -0,0 +1,116 @@
+You can obtain or create an XMPMeta object in several ways:
+
+You can use an XMPFile object to retrieve existing metadata directly from a file. The XMPFile.getXMP() method creates an XMPMeta object, which you can use to examine or modify the properties and their values. You can then use XMPFile.putXMP() to write the modified metadata back to the file.
+
+You can create an XMPMeta object with the constructor, initializing it with an XMP packet created or obtained elsewhere.
+
+You can create a new, empty XMPMeta object with the constructor, and use its methods to create entirely new namespaces and properties. You can then use XMPFile.putXMP() to inject the new metadata into a file.
+
+## Creating new metadata
+This code creates an empty XMPMeta object, uses it to set a metadata property, and serializes it to a string, which you could pass to an authoring tool, for example, or store in a file.
+
+```js
+const {XMPMeta, XMPConst} = require("uxp").xmp;
+let xmp = new XMPMeta();
+xmp.setProperty( XMPConst.NS_XMP, "CreatorTool", "My Script" );
+let xmpStr = xmp.serialize(); // Serialize the XMP packet to XML
+
+// Retrieve property
+let prop = xmp.getProperty(XMPConst.NS_XMP, "CreatorTool");
+console.log( `namespace: ${prop.namespace}, property path + name: ${prop.path}, value: ${prop.value}`);
+```
+
+## Modifying existing metadata
+This code accesses an existing XMP packet, assuming the location has been assigned to a string letiable. It sets the modification-date property to the current date and time, and stores the updated XMP packet back to the string, making it as small as possible.
+
+```js
+const { XMPMeta, XMPConst, XMPDateTime } = require("uxp").xmp;
+let xmp = new XMPMeta(xmpStr); // Object initialized with xmp packet as string
+let dateTime = new XMPDateTime(new Date()); // Now
+let oldModificationDate = xmp.getProperty(XMPConst.NS_XMP, "ModifyDate", "xmpdate");
+console.log("Old modification date: " + oldModificationDate);
+xmp.setProperty(XMPConst.NS_XMP, "ModifyDate", dateTime, "xmpdate");
+
+// Serialize to XML, in compact style
+let xmpStr = xmp.serialize(XMPConst.SERIALIZE_USE_COMPACT_FORMAT);
+```
+
+## Using XMPFile for batch processing
+This example iterates through a folder of image files and processes the metadata. The script processes each picture as follows:
+
+Reads and parses the metadata. If an image file does not contain XMP metadata, the legacy metadata is automatically converted to XMP.
+
+Deletes the list of existing creators, and adds a new creator value.
+
+Writes the modified metadata back to the file.
+
+```js
+console.log( "XMPFiles batch processing example" );
+// Load the XMPScript library
+const {XMPMeta, XMPConst, XMPFile} = require("uxp").xmp;
+
+// Iterate through the photos in the folder
```

Comment:
@kareemmyk  it would be nice to add examples of reading/writing XMP to open and embedded files in PS.

Reply:
@padmkris123 , slightly confused here.
We have a write operation in the example ``` xmp.appendArrayItem(...)``` and ```getXMP``` gives the required object to perform read.

Can you elaborate this requirement?

Reply:
AFAIU this example shows the way to set xmp on a random file. Is it also applicable to already open or photoshop docs? Vinay had shown in a demo in Partner Days that updated xmp data for an already open file. https://git.corp.adobe.com/torq/uxp-sample-plugins/tree/main/xmpSamplePs
If there is a difference in the approach between a close and open file then we should provide separate examples for each.

Reply:
Updated example for ```Photoshop Only - Writing to an active document```